tests/unittests/parsing/PreprocessorTests.cpp

```diff
@@ -1318,14 +1328,44 @@ TokenKind lexUnconnectedDrive(std::string_view text) {
 
 TEST_CASE("unconnected_drive directive") {
     CHECK(lexUnconnectedDrive("`unconnected_drive pull0") == TokenKind::Pull0Keyword);
+    CHECK_DIAGNOSTICS_EMPTY;
     CHECK(lexUnconnectedDrive("`unconnected_drive pull1") == TokenKind::Pull1Keyword);
+    CHECK_DIAGNOSTICS_EMPTY;
     CHECK(lexUnconnectedDrive("`nounconnected_drive") == TokenKind::Unknown);
     CHECK_DIAGNOSTICS_EMPTY;
 
     CHECK(lexUnconnectedDrive("`unconnected_drive asdf") == TokenKind::Unknown);
     CHECK(!diagnostics.empty());
 }
 
+bool lexCellDefine(std::string_view text) {
+    diagnostics.clear();
+
+    Preprocessor preprocessor(getSourceManager(), alloc, diagnostics);
+    preprocessor.pushSource(text);
+
+    Token token = preprocessor.next();
+    REQUIRE(token);
+    return preprocessor.getCellDefine();
+}
+
+TEST_CASE("celldefine directive") {
+    CHECK(lexCellDefine("`celldefine") == true);
+    CHECK_DIAGNOSTICS_EMPTY;
+
+    CHECK(lexCellDefine("`celldefine\n`endcelldefine") == false);
+    CHECK_DIAGNOSTICS_EMPTY;
+
+    CHECK(lexCellDefine("`endcelldefine") == false);
+    CHECK_DIAGNOSTICS_EMPTY;
+
+    CHECK(lexCellDefine("`celldefine asdf") == false);
```

Comment:
These are not an error because the celldefine handler isn't trying to parse any arguments after it. This just looks to the parser like a celldefine followed by an identifier called `asdf`

Reply:
Thanks. I was just looking up the detailed semantics of the Verilog preprocessor directives to try and figure out what's the expected behavior here.